### ISSUE TITLE
500 error: More than one row with the given identifier was found: ..., for class: rundeck.LogFileStorageRequest

### ISSUE THREAD

**Issue by gschueler** [rundeck/rundeck#5081]
**Describe the bug**

500 error occurs on activity page, with message 

> More than one row with the given identifier was found: XXX, for class: rundeck.LogFileStorageRequest

(XXX is an ID number)

**My Rundeck detail**

* Rundeck version: 3.x

**To Reproduce**

Inconsistent, possibly aborting an execution right as it also finishes normally, or doing so from another cluster member.

**Expected behavior**

No error

**Additional context**

Execution declare `hasOne` for LogFileStorageRequest, however there is no  `unique` constraint applied to the associate.  Gorm 6.1 declares [`it's a good idea to add a unique constraint on one side of the one-to-one relationship`](http://gorm.grails.org/6.1.x/hibernate/manual/index.html#manyToOneAndOneToOne) but does not mention any consequence of not doing so.  The unique constraint should be applied, and the LogFileStorageService should handle failure to create the request if it already exists for an execution.

**Comment by G3NSVRV** [comment]
+1

**Comment by G3NSVRV** [comment]
Also,
Can be workaround by deleting the orphaned or duplicated execution ID on the Database:

```
select * from log_file_storage_request where id=$executionID
select * from log_file_storage_request where execution_id=$executionID
select * from execution where id=$executionID
```

The result should be something like:
(ID: 1372108)

```
id |version|plugin_name |date_created |filetype|last_updated |completed|execution_id|
-——|———-|——————————-|—————————-|————|—————————-|————-|——————|
1372071| 1|org.rundeck.amazon-s3|2019-05-14 09:17:51| |2019-05-14 09:17:52| 1| 1372108|
1372072| 1|org.rundeck.amazon-s3|2019-05-14 09:17:52| |2019-05-14 09:17:52| 1| 1372108|
```

We can delete the ID=1372071 or ID=1372072, and then try again.

This issue commonly happens with amazon-s3 log storage plugin

**Comment by fgutierrezz** [comment]
+1 similar errors found after upgrade to 3.2.2 , also using s3 log storage

**Comment by elioe** [comment]
Rundeck version 3.1.2
Duplicated executions were found, but, using Azure log storage plugin, a weird behavior of the project that owns those executions could be watched, after deleting one of the dups, the correct function of Rundeck come again.

**Comment by fdevans** [comment]
We will continue to work on this fix for 3.2.7.  If you are experiencing this issue we will post some further instructions to address it manually for now.

**Comment by fdevans** [comment]
If you are experiencing this issue here are some steps to address:

Manually run this SQL against your Rundeck database.

```sql
select count(id) from log_file_storage_request
GROUP BY execution_id HAVING count(execution_id) > 1;
```

If a row count comes back as 0 you can continue with the upgrade and skip the following steps.

If a row count greater than 0 is returned, the duplicates must be removed:

**Locate duplicates**
```sql
DELETE FROM log_file_storage_request
WHERE id IN (
    SELECT * FROM (
        SELECT log.id FROM log_file_storage_request log
        JOIN (
            SELECT executiuon_id
            FROM log_file_storage_request
            GROUP BY execution_id
            HAVING count(*) > 1
        ) dupe
        ON log.execution_id = dupe.execution_id
        WHERE log.completed=0
    ) uncompleted_dupes
);
```

**Delete all but one duplicate entry**
```sql
DELETE FROM log_file_storage_request
WHERE id NOT IN (
    SELECT * FROM (
        SELECT min(id)
        FROM log_file_storage_request
        GROUP BY execution_id
    ) dupes
);
```
